src/components/Firestore/Requests/RequestPath/index.tsx

```diff
@@ -0,0 +1,170 @@
+  fullRequestPath: string,
+  pathContainerWidth?: number,
+  // Props used for testing purposes only
+  mockedPathOffsetWidth?: number,
+  mockedIconOffsetWidth?: number
+): void {
+  const pathHtmlElement = pathTextElement.current;
+  if (!pathHtmlElement || !pathContainerWidth) {
+    return;
+  }
+  const pathTextString = pathHtmlElement.textContent;
+  // Use the mocked offsetWidths for testing, otherwise use the real values
+  // NOTE: using this mocked values is important because jsdom does not handle
+  // layouts (which means that every variable that contains a size measurement
+  // will be equal to 0 during a test)
+  const pathTextWidth = mockedPathOffsetWidth || pathHtmlElement.offsetWidth;
+  const copyIconButtonWidth =
+    mockedIconOffsetWidth || copyButtonElement.current?.offsetWidth || 0;
+  // Calculate the width in px of a single character: (totalWidth / totalCharacters)
+  // NOTE: Even though the font-family is not monospace (every character may have a different width),
+  // this solution is accepted because the pixel approximation varies by only decimals.
+  const totalPathCharacters = pathTextString?.length || 0;
+  const pathCharacterPxWidth = totalPathCharacters
+    ? pathTextWidth / totalPathCharacters
+    : 0;
+  // End truncation to avoid x/0 division
+  if (!pathCharacterPxWidth) {
+    return;
+  }
+  // Calculate amount of characters that fit inside the pathHtmlElement: (totalWidth / widthPerCharacter)
+  // NOTE: Math.floor is chosen over Math.ceil to ensure there is a small space
+  // between the request-path and the copy-button
+  const pathHtmlElementAvailableWidth =
+    pathContainerWidth - copyIconButtonWidth;
+  const newPathMaxCharacters = Math.floor(
+    pathHtmlElementAvailableWidth / pathCharacterPxWidth
+  );
+  // Calculate where to start the truncated substring
+  const newPathStart = fullRequestPath.length - newPathMaxCharacters;
+  // Truncate the path only if the full path doesn't fit in the HTMLElement,
+  // otherwise just set the new path-string to be the full request-path.
+  // NOTE: If the path will be truncated, a '...' substring is prepended (added at the begining
+  // of the path) and a (+ 3) is added to the substring initial-index to compensate
+  // the space of the '...' characters ('...' replaces the first 3 characters of the truncated path)
+  const newTruncatedPathString =
+    newPathStart > 0
+      ? `...${fullRequestPath.substr(newPathStart + 3)}`
+      : fullRequestPath;
+  // Only update the HTMLElement's inner-text if the string value changed
+  // after truncation (to avoid unnecessary rerenders of component)
+  if (pathHtmlElement.textContent !== newTruncatedPathString) {
+    pathHtmlElement.textContent = newTruncatedPathString;
+  }
+}
+
+interface Props {
+  resourcePath: string;
+  setShowCopyNotification: (value: boolean) => void;
+  requestPathContainerWidth?: number;
+  // Props used for testing purposes only
+  mockedPathOffsetWidth?: number;
```

Comment:
I'm not super excited about adding a couple of properties whose only use is testing. See my other comment on the sizing function